### ISSUE TITLE
DataGrid.SaveLayout() and .RestoreLayout()

### ISSUE THREAD

**Issue by MichaelSvejcar** [AvaloniaUI/Avalonia#12985]
Hello,
it would be really nice to have the DataGrid.SaveLayout() and .RestoreLayout() methods implemented, which would de/serialize it's layout to stream/file. For Avalonia v0.10, I know that the [Np.Avalonia.Visuals package](https://www.codeproject.com/Articles/5329865/Avalonia-DataGrid-Advanced-Features-coming-from-NP)  was doing exactly this, but I was not able to make run it with Avalonia 11 anymore.

**Comment by timunie** [comment]
I don't think that this is straight forward enough to implement in Avalonia directly. Due to:

- where to save it? 
- Each plaform handles it different 
- which DataFormat?
- Which Serializer?
- ...

Better to wait for 3rd party lib to update or implement it on your side (my personal opinion)

**Comment by TETYYS** [comment]
> * where to save it?

Should be consumers problem, SaveLayout would return the result

> * Each plaform handles it different

Are you thinking of saving to stream/file part? DataGrid is Avalonia-native, so I don't see a problem here

> * which DataFormat?
> * Which Serializer?

Doesn't matter as long as RestoreLayout(x) works where x is the input of SaveLayout. SaveLayout could also return an opaque serializable object if there is no serializer in Avalonia

**Comment by timunie** [comment]
I guess this all can also be done using a Behavior, which gives much more flexibility what to save and what not.
